### DISCUSSION TITLE
Unable to Push files/folders into new GitHub repository

### DISCUSSION THREAD

**Discussion by ThomasL08** [community/community#23877] [New to GitHub]
<p>Hello, like the title suggests I’m unable to push files/folders into newly created github repository or any repository I have tried setting up a SSN key it it never connect to  github using the command to test connection ssh -T <a href="mailto:git@github.com">git@github.com</a> or using the https method setting up a personal access token it keep deny me access does anyone know whats going on and how to fix this issue?<br>
Thank you.</p>

**Comment by Ifycode** [comment]
<p>Hey, what OS do you use? And how did you go about setting up the ssh?<br>
Also send a screenshot of the error you are getting in your terminal.</p>

**Comment by ThomasL08** [comment]
<p>Hi I’m using chrome os with linux enable so I guess you say linux and I followed the steps here<br>
<a href="https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent" rel="noopener nofollow ugc">Github Doc for Linux</a><br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/7/17de8d0d129b25419e8145c7e29afd24bf121379.png" data-download-href="/uploads/short-url/3p9NilZDVgI40QZJvqS4Gs23Tn3.png?dl=1" title="Screenshot 2021-09-06 10.11.28 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/80377808/181079802-82b273b0-b0d7-4470-802d-649e6643c963.png" alt="Screenshot 2021-09-06 10.11.28 PM" data-base62-sha1="3p9NilZDVgI40QZJvqS4Gs23Tn3" width="690" height="133" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/7/17de8d0d129b25419e8145c7e29afd24bf121379_2_690x133.png, https://aws1.discourse-cdn.com/github/original/3X/1/7/17de8d0d129b25419e8145c7e29afd24bf121379.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/7/17de8d0d129b25419e8145c7e29afd24bf121379.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/7/17de8d0d129b25419e8145c7e29afd24bf121379_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2021-09-06 10.11.28 PM</span><span class="informations">798×154 35.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment]
<p>That error message looks like GitHub doesn’t know your SSH key. After creating the key (as described in the documentation link you posted) you need to add it to your account, see here: <a href="https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account" class="inline-onebox">Adding a new SSH key to your GitHub account - GitHub Docs</a></p>

**Comment by Ifycode** [comment]
<p>Ok. I’m not used to linux (just windows and mac). But also check out the part of github documentation that talks about your issue. Let’s see if this helps solve it.</p>
<h1>
<a name="error-permission-denied-publickey-1" class="anchor" href="#error-permission-denied-publickey-1"></a>Error: Permission denied (publickey)</h1>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/github/authenticating-to-github/troubleshooting-ssh/error-permission-denied-publickey">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/github/authenticating-to-github/troubleshooting-ssh/error-permission-denied-publickey" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/45185388/181079804-d702f639-2b5b-4561-9eac-b43b15d533da.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/github/authenticating-to-github/troubleshooting-ssh/error-permission-denied-publickey" target="_blank" rel="noopener nofollow ugc">Error: Permission denied (publickey) - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by ThomasL08** [comment]
<p>I have created and connected to GitHub in the setting section where saying SSH key and I checked the fingerprint it’s the same. However, the problem still persists.</p>

**Comment by ThomasL08** [comment, marked as answer]
<p>can someone walk me through how to do this part?</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/5/45d3910ede3871f28f6ca59e8eab40b5420d65b4.png" data-download-href="/uploads/short-url/9XIdWrF5hvZ9IWU8sqCIKKOYkGU.png?dl=1" title="Screenshot 2021-09-08 105156" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/80377808/181079805-79ace613-9ba0-4d06-b17c-b673f9304082.png" alt="Screenshot 2021-09-08 105156" data-base62-sha1="9XIdWrF5hvZ9IWU8sqCIKKOYkGU" width="688" height="499" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/5/45d3910ede3871f28f6ca59e8eab40b5420d65b4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2021-09-08 105156</span><span class="informations">779×566 26.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
I put the eval agent and got the output of agent pid then numbers but how do I do the next part?<br>
If I named the file as testkey.pub and used the legacy version to generate SSH key what would be the command <code>ssh-add ~/.ssh/id_rsa/testkey.pub</code><br>
Thank you.

**Comment by airtower-luna** [comment]
<p><code>ssh-add</code> needs the <em>private</em> key, not the public one. If you have a <code>testkey.pub</code> file with your public key, there should be a matching <code>testkey</code> file holding the private one.</p>
